pydantic/_internal/_generate_schema.py

```diff
@@ -1153,15 +1129,28 @@ def _annotated_schema(self, annotated_type: Any) -> core_schema.CoreSchema:
         Generate schema for an Annotated type, e.g. `Annotated[int, Field(...)]` or `Annotated[int, Gt(0)]`.
         """
         first_arg, *other_args = get_args(annotated_type)
-        return self._apply_annotations(CallbackGetCoreSchemaHandler(self.generate_schema), first_arg, other_args)
 
-    def _get_prepare_pydantic_annotations_for_known_type(self, obj: Any) -> Callable[..., Any] | None:
-        from decimal import Decimal
+        source_type, annotations = self._prepare_annotations(first_arg, other_args)
 
+        return self._apply_annotations(
+            CallbackGetCoreSchemaHandler(self.generate_schema, self.generate_schema), source_type, annotations
+        )
+
+    def _get_prepare_pydantic_annotations_for_known_type(
+        self, obj: Any, annotations: tuple[Any, ...]
+    ) -> Iterable[Any] | None:
         from . import _std_types_schema as std_types
 
-        if obj is Decimal:
-            return std_types.decimal_prepare_pydantic_annotations
+        for gen in (
+            std_types.decimal_prepare_pydantic_annotations,
+            std_types.sequence_like_prepare_pydantic_annotations,
+            std_types.datetime_prepare_pydantic_annotations,
+            std_types.uuid_prepare_pydantic_annotations,
+            std_types.path_schema_prepare_pydantic_annotations,
+        ):
```

Comment:
I think we should collapse these into a single `std_types.match_to_known_type` or something that does this for loop internally. But open to other ideas for how to organize this.